### PR TITLE
Add a controller to manage contracts

### PR DESCRIPTION
**Pull request by oleg1std**

### CODE REVIEW

File: HappyTravel.Hiroshima.Data/DirectContractsDbContext.cs [hunk 1]
```diff
@@ -90,8 +125,10 @@ private void AddAccommodations(ModelBuilder modelBuilder)
                 e.Property(a => a.CheckInTime);
                 e.Property(a => a.CheckOutTime);
                 e.Property(a => a.OccupancyDefinition).HasColumnType("jsonb");
+                e.Property(a => a.UserId).IsRequired();
                 e.HasIndex(a=> a.Coordinates).HasMethod("GIST");
                 e.HasOne<Models.Location.Location>().WithMany().HasForeignKey(a=> a.LocationId).IsRequired();
+                e.HasOne<User>().WithMany().HasForeignKey(a=> a.UserId);
```
Comment (hasmikharutyunyan) [inline HappyTravel.Hiroshima.Data/DirectContractsDbContext.cs:131]: As I remember we decided , not to use foreign keys

File: HappyTravel.Hiroshima.DirectContracts/Services/Management/AccommodationManagementService.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        }
+        
+        
+        public async Task<Accommodation> GetAccommodation(int userId, int accommodationId) 
+            => await _dbContext.Accommodations.SingleOrDefaultAsync(a => a.Id == accommodationId && a.UserId == userId);
+
+        
+        public async Task<Accommodation> AddAccommodation(Accommodation accommodation)
+        { 
+            var entry = await _dbContext.Accommodations.AddAsync(accommodation);
```
Comment (hasmikharutyunyan) [inline HappyTravel.Hiroshima.DirectContracts/Services/Management/AccommodationManagementService.cs:22]: why nod just Add ?

File: HappyTravel.Hiroshima.DirectManager/Services/ContractManagementService.cs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                AccommodationId = accommodationId,
+                Name = contract.Name,
+                Description = contract.Description,
+                ValidFrom = contract.ValidFrom,
+                ValidTo = contract.ValidTo
+            };
+        
+        
+        private readonly IUserContextService _userContext;
+        private readonly DirectContracts.Services.Management.IContractManagementService _contractManagement;
```
Comment (hasmikharutyunyan) [inline HappyTravel.Hiroshima.DirectManager/Services/ContractManagementService.cs:160]: I think with postfix 'service' will b better

File: HappyTravel.Hiroshima.DirectContracts/Services/Management/ContractManagementService.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                (contractAccommodationRelation, accommodation) => new ContractAccommodationRelationAndAccommodation
+                {
+                    ContractAccommodationRelation = contractAccommodationRelation, Accommodation = accommodation
+                });
+        
+        
+        private readonly DirectContractsDbContext _dbContext;
+
+        
+        class ContractAccommodationRelationAndAccommodation
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectContracts/Services/Management/ContractManagementService.cs:111]: An access modifier is missed

File: HappyTravel.Hiroshima.DirectManager/Services/ContractManagementService.cs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+using HappyTravel.Hiroshima.Data.Models;
+using HappyTravel.Hiroshima.DirectManager.Infrastructure;
+
+namespace HappyTravel.Hiroshima.DirectManager.Services
+{
+    public class ContractManagementService: IContractManagementService
+    {
+        public ContractManagementService(IUserContextService userContextService, DirectContracts.Services.Management.IContractManagementService contractManagement, DirectContracts.Services.Management.IAccommodationManagementService accommodationManagement)
+        {
+            _userContext = userContextService;
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectManager/Services/ContractManagementService.cs:15]: Those names are against the guideline

File: HappyTravel.Hiroshima.DirectManager/Services/ContractManagementService.cs [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+        
+        private async Task<bool> IsUserAccommodation(int userId, int accommodationId)
+        {
+            var accommodation = await _accommodationManagement.GetAccommodation(userId, accommodationId);
+            return !(accommodation is null);
+        }
+        
+        
+        private async Task<bool> IsUserContract(int userId, int contractId)
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectManager/Services/ContractManagementService.cs:140]: The name is pointless

File: HappyTravel.Hiroshima.DirectManager/Services/ContractManagementService.cs [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+        {
+            var contract = await _contractManagement.GetContract(userId, contractId);
+            return !(contract is null);
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectManager/Services/ContractManagementService.cs:143]: The check is way strange

File: HappyTravel.Hiroshima.DirectContracts/Services/Management/ContractManagementService.cs [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+            contractEntry.State = EntityState.Detached;
+        }
+
+        
+        public async Task DeleteContract(int userId, int contractId)
+        {
+            await DeleteContractAccommodationRelations(contractId);
+
+            var contract = _dbContext.Contracts.SingleOrDefault(c => c.Id == contractId && c.UserId == userId);
+            if (!(contract is null))
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectContracts/Services/Management/ContractManagementService.cs:57]: Could be rewritten simpler

File: HappyTravel.Hiroshima.DirectContracts/Services/Management/ContractManagementService.cs [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+using HappyTravel.Hiroshima.Data;
+using HappyTravel.Hiroshima.Data.Models;
+using HappyTravel.Hiroshima.Data.Models.Accommodations;
+using Microsoft.EntityFrameworkCore;
+
+namespace HappyTravel.Hiroshima.DirectContracts.Services.Management
+{
+    public class ContractManagementService : IContractManagementService
```
Comment (kirillta) [inline HappyTravel.Hiroshima.DirectContracts/Services/Management/ContractManagementService.cs:11]: This class looks like a repository over a repository. Doubt we need it.

**Pull request merged**